enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -1353,22 +1484,33 @@ func (e *partitionEvictor) resampleK(k int) error {
 	iter := e.iter()
 	defer iter.Close()
 
+	// read new entries to put in the pool.
+	additions := make([]*evictionPoolEntry, 0, k)
 	for i := 0; i < k; i++ {
 		entries, err := e.randomSample(iter, sampleN)
 		if err != nil {
 			return err
 		}
-		for _, entry := range entries {
-			e.samplePool = append(e.samplePool, entry)
-		}
+		additions = append(additions, entries...)
+
 	}
 
+	filtered := e.samplePool[:0]
```

Comment:
can you make a new slice here instead of reusing samplePool? appending to filtered will modify the array backing samplePool which you are iterating over.

Reply:
Done.